File: 프로그래머스/1/42840. 모의고사/모의고사.kt
class Solution {
    fun solution(answers: IntArray): IntArray {
        val answer = mutableListOf<Int>()
        
        val soopojaScore = mutableListOf(0, 0, 0)
        val soopojaPattern = arrayOf(
            intArrayOf(1, 2, 3, 4, 5),
            intArrayOf(2, 1, 2, 3, 2, 4, 2, 5),
            intArrayOf(3, 3, 1, 1, 2, 2, 4, 4, 5, 5)
        )
        
        for ((i, a) in answers.withIndex()) {
            for ((j, p) in soopojaPattern.withIndex()) {
                if (a == p[i % p.size]) {
                    soopojaScore[j] += 1
                }
            }
        }
        
        val max_score = soopojaScore.maxOrNull()
        
        for ((i, score) in soopojaScore.withIndex()) {
            if (score == max_score) {
                answer.add(i + 1)
            }
        }
        
        return answer.sorted().toIntArray()
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    p1 = [1, 2, 3, 4, 5]
    p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    n1, n2, n3 = len(p1), len(p2), len(p3)
    cnt = [0] * 3
    
    for i, ans in enumerate(answers):
        if ans == p1[i % n1]:
            cnt[0] += 1
        if ans == p2[i % n2]:
            cnt[1] += 1
        if ans == p3[i % n3]:
            cnt[2] += 1
            
    max_cnt = max(cnt)
    
    for i, c in enumerate(cnt):
        if max_cnt == c:
            answer.append(i + 1)
    
    return answer